src/Parsers/IParser.h

```diff
@@ -25,10 +25,10 @@ namespace ErrorCodes
 struct Expected
 {
     const char * max_parsed_pos = nullptr;
-    std::set<const char *> variants;
+    std::vector<const char *> variants;
```

Comment:
Does it add duplicates to the error messages?

Reply:
You are right. I was expecting the same token / parser to not be attempted more than once but that's not true. I'll compare vector+find vs set.